### PR TITLE
feat: add ManagerPanelPage component

### PR DESCRIPTION
**Pull request by ike-tom**

### CODE REVIEW

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Container from '@mui/material/Container';
+import IconButton from '@mui/material/IconButton';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import Toolbar from '@mui/material/Toolbar';
+import Tooltip from '@mui/material/Tooltip';
+import Typography from '@mui/material/Typography';
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: All those imports from `@mui/material/...` cam be merged

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    anchorElUser,
+    handleOpenNavMenu,
+    handleOpenUserMenu,
+    handleCloseNavMenu,
+    handleCloseUserMenu,
+  } as const;
+};
+
+export const ManagerPanelPage = () => {
+  const { anchorElNav, anchorElUser, handleOpenNavMenu, handleOpenUserMenu, handleCloseNavMenu, handleCloseUserMenu } =
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: Add empty line for readability

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+              anchorOrigin={{
+                vertical: 'top',
+                horizontal: 'right',
+              }}
+              keepMounted
+              transformOrigin={{
+                vertical: 'top',
+                horizontal: 'right',
+              }}
+              open={Boolean(anchorElUser)}
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: You can move this `Boolean(anchorElUser)` to ` const` called for example `isUserMenuOpen` and move it then to the `customHook` you created

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+              anchorOrigin={{
+                vertical: 'bottom',
+                horizontal: 'left',
+              }}
+              keepMounted
+              transformOrigin={{
+                vertical: 'top',
+                horizontal: 'left',
+              }}
+              open={Boolean(anchorElNav)}
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: You can move this `Boolean(anchorElNav)` to ` const` called for example `isNavigationMenuOpen` and move it then to the `customHook` you created

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+import Toolbar from '@mui/material/Toolbar';
+import Tooltip from '@mui/material/Tooltip';
+import Typography from '@mui/material/Typography';
+import { MouseEvent, useState } from 'react';
+
+const pages = ['Orders', 'Menu', 'Tables', 'Reports'];
+const settings = ['Add new meal', 'Add new employee', 'Logout'];
+
+const useManagerPanelPage = () => {
+  const [anchorElNav, setAnchorElNav] = useState<null | HTMLElement>(null);
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: Somehow, for me those `anchorElNav` and `anchorElUser` are not verbose enough. Maybe call them `anchorNavigationMenuEl`  and `anchorUserMenuEl` for readability?

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                <MenuItem key={page} onClick={handleCloseNavMenu}>
+                  <Typography textAlign="center">{page}</Typography>
+                </MenuItem>
+              ))}
+            </Menu>
+          </Box>
+          <Box sx={{ flexGrow: 1, display: { xs: 'none', md: 'flex' } }}>
+            {pages.map((page) => (
+              <Button key={page} onClick={handleCloseNavMenu} sx={{ my: 2, color: 'white', display: 'block' }}>
+                {page}
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: I don't understand this part a bit. What is the point of displaying in `Menu` the pages `Typography` component and making another map of `pages` to render `Button` components that do the same thing?

**Review by ruljin — CHANGES_REQUESTED** [review]

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx (hunk 6 above)
Comment (ike-tom) [reply src/components/ManagerPanelPage/ManagerPanelPage.tsx]: One box is for mobile views and the other is for desktop.

Comment (ruljin) [reply src/components/ManagerPanelPage/ManagerPanelPage.tsx]: Okay and is there any reason why the same component cant be used for it. It is how usually it is done

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  IconButton,
+  Menu,
+  MenuItem,
+  Toolbar,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+import { MouseEvent, useState } from 'react';
+
+const pages = ['Orders', 'Menu', 'Tables', 'Reports'];
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: Use `SCREAMING_SNAKE_CASE`

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+const settings = ['Add new meal', 'Add new employee', 'Logout'];
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: Use `SCREAMING_SNAKE_CASE`

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+                </MenuItem>
+              ))}
+            </Menu>
+          </Box>
+        </Toolbar>
+      </Container>
+    </AppBar>
+  );
+};
+export default ManagerPanelPage;
```
Comment (ruljin) [inline src/components/ManagerPanelPage/ManagerPanelPage.tsx]: That `export default` is redundant. You already export the component itself here"

`export const ManagerPanelPage ...`

**Review by ruljin — CHANGES_REQUESTED** [review]

File: src/components/ManagerPanelPage/ManagerPanelPage.tsx (hunk 6 above)
Comment (ike-tom) [reply src/components/ManagerPanelPage/ManagerPanelPage.tsx]: TBH, I don't know how to fix this issue. I tried to merge it into one component, but I can not style it properly. Every time something goes wrong, and it does not look like I want it to look.

File: src/hooks/useRoute.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import { ROUTES } from '../routes/Routes';
+
+export const useRoute = () => {
```
Comment (ruljin) [inline src/hooks/useRoute.ts]: TBH right now I think it should be called `useRoutes` :D

**Review by ruljin — CHANGES_REQUESTED** [review]

**Review by ruljin — APPROVED** [review]

**Pull request merged**